src/components/ec-inline-input-field/ec-inline-input-field.spec.js

```diff
@@ -140,4 +158,87 @@ describe('EcInlineInputField', () => {
       });
     });
   });
+
+  describe('when component is copiable', () => {
+    it('should throw an error if the tooltip props were not given', () => {
+      withMockedConsole((errorSpy) => {
+        mountInlineInputField(
+          {
+            isEditable: false,
+            isCopiable: true,
+          },
+        );
+        expect(errorSpy).toHaveBeenCalledTimes(2);
+        expect(errorSpy.mock.calls[0][0]).toContain('Invalid prop: type check failed for prop "tooltipTextSuccess"');
+        expect(errorSpy.mock.calls[1][0]).toContain('Invalid prop: type check failed for prop "tooltipTextError"');
+      });
+    });
+
+    it('should render as expected', async () => {
+      const wrapper = mountInlineInputField(
+        {
+          isEditable: false,
+          isCopiable: true,
+          tooltipTextSuccess,
+          tooltipTextError,
+        },
+      );
+      await wrapper.vm.$nextTick();
+
+      expect(wrapper.element).toMatchSnapshot();
+      expect(wrapper.findByDataTest('ec-inline-input-field-copy').exists()).toBeTruthy();
+    });
+
+    it('should show the success tooltip after succesfully triggering the copy method', async () => {
+      mockClipboardCopySuccess();
```

Comment:
This does make the promise to be resolved and enter inside "then" where the value of "isCopied" changes from "false" to "true". If console log it we can see this is the case. The tooltip though during testing doesn't see this change. Similar issue with the error tooltip which even though the test pass is a false positive.

Reply:
It is not updating because the mock for the tooltip doesn't update its values. It's simple, it only has `bind` so it works only if component is mounted. For updates you have to change the mock:
```
jest.mock('../../directives/ec-tooltip', () => {
  const MockedTooltipDirective = {
    bind(el, { value }) {
      if (value.content) {
        el.setAttribute('mocked-tooltip-content', value.content);
        el.setAttribute('mocked-tooltip-classes', value.classes);
      }
    },
    update(el, { value }) {
      if (value.content) {
        el.setAttribute('mocked-tooltip-content', value.content);
        el.setAttribute('mocked-tooltip-classes', value.classes);
      }
    },
  };
  return MockedTooltipDirective;
});
```